zoom_rooms/indico_zoom_rooms/models.py

```diff
@@ -0,0 +1,137 @@
+        'title': vc_room.name,
+        'url': vc_room.data['url'],
+    }
+
+
+class ZoomRoomsQueueEntry(db.Model):
+    """Pending calendar updates"""
+
+    __tablename__ = 'queue'
+    __table_args__ = (
+        db.CheckConstraint(
+            f'action != {ZoomRoomsAction.delete} OR (entry_data IS NULL AND extra_args IS NULL)', 'delete_has_no_args'
+        ),
+        db.CheckConstraint(f'action = {ZoomRoomsAction.delete} OR entry_data IS NOT NULL', 'other_actions_have_args'),
+        db.CheckConstraint(f'action = {ZoomRoomsAction.move} OR extra_args IS NULL', 'move_has_extra_args'),
+        db.CheckConstraint(
+            f'action = {ZoomRoomsAction.move} OR extra_args IS NULL', 'other_actions_have_no_extra_args'
+        ),
+        {'schema': 'plugin_zoom_rooms'},
+    )
+    #: Entry ID (mainly used to sort by insertion order)
+    id = db.Column(db.Integer, primary_key=True)
+    #: ID of the Entry (to be used on the server side)
+    entry_id = db.Column(db.String, nullable=False)
+    #: ID of the Zoom Room
+    zoom_room_id = db.Column(db.String, nullable=False)
+    #: :class:`ZoomRoomsAction` to perform
+    action = db.Column(PyIntEnum(ZoomRoomsAction), nullable=False)
+    #: The actual entry's data
+    entry_data: EntryData = db.Column(
+        JSONB(none_as_null=True),
+    )
+    #: Additional args to be sent with the request
+    extra_args: ExtraArgs = db.Column(
+        JSONB(none_as_null=True),
+    )
+
+    def __repr__(self):
+        action = ZoomRoomsAction(self.action).name
+        return format_repr(self, 'id', 'entry_id', action=action)
+
+    @classmethod
+    def record(
+        cls,
+        action: int,
+        zoom_room_id: str,
+        assoc: VCRoomEventAssociation | None = None,
+        obj: Event | Contribution | SessionBlock | None = None,
+        vc_room: VCRoom | None = None,
+        args: OperationArgs | None = None,
+    ):
+        """Record in the database a ZoomRoomsQueueEntry based on an action and its parameters."""
+        if obj is None and assoc is not None:
+            obj = assoc.link_object
+            vc_room = vc_room or assoc.vc_room
+        elif assoc is None and (obj is None or vc_room is None):
+            raise ValueError('Either assoc or obj + vc_room must be provided')
+
+        args = args or {}
+
+        if new_zr_id := args.pop('new_zr_id', None):
+            extra_args = {'new_zr_id': new_zr_id}
+        else:
+            extra_args = None
+
+        entry = cls(
+            action=action,
+            entry_id=make_zoom_room_entry_id(zoom_room_id, obj, vc_room),
+            # DELETE operations have no additional parameters, only the ID
+            entry_data=None if action == ZoomRoomsAction.delete else EntryData(get_entry_data(obj, vc_room), **args),
+            extra_args=extra_args,
+            zoom_room_id=zoom_room_id,
+        )
+        db.session.add(entry)
+        db.session.flush()
+
+    @property
+    def data(self) -> dict:
```

Comment:
Am I blind or is this used ONLY in the tests? If yes, I'd rather make this a `_entry_data` function in the in the file containing the tests, instead of masking it part of the actual model. It feels dirty to have code used only in tests...

Reply:
It is. I kept it because it might be used in the future. But I agree with you, let's move it to a less visible place.